storage/database/db_manager_test.go

```diff
@@ -672,24 +673,207 @@ func TestDBManager_Governance(t *testing.T) {
 	// TODO-Klaytn-Database Implement this!
 }
 
-func TestDBManager_StateTrieMigration(t *testing.T) {
+func TestDatabaseManager_CreateMigrationDBAndSetStatus(t *testing.T) {
+	for i, dbm := range dbManagers {
+		if dbConfigs[i].DBType == MemoryDB {
+			continue
+		}
+
+		// check if migration fails on single DB
+		if !dbm.IsPartitioned() {
+			migrationBlockNum := uint64(12345)
+
+			// check if not in migration
+			assert.False(t, dbManagers[i].InMigration(), "migration status should be not set before testing")
+
+			// check if create migration fails
+			err := dbm.CreateMigrationDBAndSetStatus(migrationBlockNum)
+			assert.Error(t, err, "error expected on non-partitioned DB") // expect error
+
+			continue
+		}
+
+		// check if migration fails when in migration
+		{
+			migrationBlockNum := uint64(34567)
+
+			// check if not in migration
+			assert.False(t, dbManagers[i].InMigration(), "migration status should be not set before testing")
+
+			// set migration status
+			dbm.setStateTrieMigrationStatus(migrationBlockNum)
+			assert.True(t, dbManagers[i].InMigration())
+
+			// check if create migration fails
+			err := dbm.CreateMigrationDBAndSetStatus(migrationBlockNum)
+			assert.Error(t, err, "error expected when in migration state") // expect error
+
+			// reset migration status for next test
+			dbm.setStateTrieMigrationStatus(0)
+		}
+
+		// check if CreateMigrationDBAndSetStatus works as expected
+		{
+			migrationBlockNum := uint64(56789)
+
+			// check if not in migration state
+			assert.False(t, dbManagers[i].InMigration(), "migration status should be not set before testing")
+
+			err := dbm.CreateMigrationDBAndSetStatus(migrationBlockNum)
+			assert.NoError(t, err)
+
+			// check if in migration state
+			assert.True(t, dbm.InMigration())
+
+			// check migration DB path in MiscDB
+			migrationDBPathKey := append(databaseDirPrefix, common.Int64ToByteBigEndian(uint64(StateTrieMigrationDB))...)
+			fetchedMigrationPath, err := dbm.getDatabase(MiscDB).Get(migrationDBPathKey)
+			assert.NoError(t, err)
+			expectedMigrationPath := "statetrie_migrated_" + strconv.FormatUint(migrationBlockNum, 10)
```

Comment:
```suggestion
			expectedMigrationPath := dbBaseDirs[StateTrieMigrationDB] + strconv.FormatUint(migrationBlockNum, 10)
```

Reply:
I changed to have less logic in expected values as @aidan-kwon commented above. I cannot decide which one to choose because both comments have a point. 😢